libs/server/Resp/Bitmap/BitmapCommands.cs

```diff
@@ -432,10 +431,73 @@ private bool StringBitPosition<TGarnetApi>(byte* ptr, int count, ref TGarnetApi
         /// <summary>
         /// Performs bitwise operations on multiple strings and store the result.
         /// </summary>
-        private bool StringBitOperation<TGarnetApi>(int count, byte* ptr, BitmapOperation bitop, ref TGarnetApi storageApi)
+        private bool StringBitOperation<TGarnetApi>(int count, byte* ptr, ref TGarnetApi storageApi)
             where TGarnetApi : IGarnetApi
         {
+            // <AND | OR | XOR | NOT> destkey key
+            if (count < 3)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_UNK_CMD, ref dcurr, dend))
+                    SendAndReset();
+
+                return true;
+            }
+
+            Span<byte> op = default;
+            if (!RespReadUtils.ReadSpanByteWithLengthHeader(ref op, ref ptr, recvBufferPtr + bytesRead))
+            {
+                return false;
+            }
+
+            readHead = (int)(ptr - recvBufferPtr);
+
+            count--;
+
+            BitmapOperation bitop;
+            if (op.SequenceEqual(CmdStrings.AND) || op.SequenceEqual(CmdStrings.and))
```

Comment:
Matching of the operation type should happen at parsing layer. I think we treat this as subcommand plus for parameters that are fixes and few bytes it might be better to match them using MemoryMarshal.Read